app/src/main/java/com/cornellappdev/android/eatery/model/EateryModel.java

```diff
@@ -0,0 +1,234 @@
+    String payMethodsString = "Pay Methods: " + mPayMethods.toString();
+    String menuString = "";
+
+    return info + "\n" + locationString + "\n" + payMethodsString + "\n" + "Menu" + "\n"
+        + menuString;
+  }
+
+  public abstract List<String> getMealItems();
+
+  public boolean isOpen() {
+    return getCurrentStatus() == Status.OPEN;
+  }
+
+  public abstract Status getCurrentStatus();
+
+  public String getName() {
+    return mName;
+  }
+
+  public void setName(String name) {
+    this.mName = name;
+  }
+
+  public String getNickName() {
+    return mNickName;
+  }
+
+  public void setNickName(String nickName) {
+    this.mNickName = nickName;
+  }
+
+  public boolean matchesFilter() {
+    return mMatchesFilter;
+  }
+
+  public void setMatchesFilter(boolean yesOrNo) {
+    mMatchesFilter = yesOrNo;
+  }
+
+  public boolean matchesSearch() {
+    return mMatchesSearch;
+  }
+
+  public void setMatchesSearch(boolean yesOrNo) {
+    mMatchesSearch = yesOrNo;
+  }
+
+  public List<String> getPayMethods() {
+    return mPayMethods;
+  }
+
+  public void setPayMethods(List<String> paymentMethods) {
+    this.mPayMethods = paymentMethods;
+  }
+
+  public String getBuildingLocation() {
+    return mBuildingLocation;
+  }
+
+  public void setBuildingLocation(String buildingLocation) {
+    this.mBuildingLocation = buildingLocation;
+  }
+
+  public int getId() {
+    return id;
+  }
+
+  public void setId(int id) {
+    this.id = id;
+  }
+
+  public boolean isHardCoded() {
+    return isHardCoded;
+  }
+
+  public void setHardCoded(boolean hardCoded) {
+    isHardCoded = hardCoded;
+  }
+
+  public LatLng getLatLng() {
+    return new LatLng(lat, lng);
+  }
+
+  public void setLatLng(LatLng latlng) {
+    this.lat = latlng.latitude;
+    this.lng = latlng.longitude;
+  }
+
+  public boolean isOpenPastMidnight() {
+    return openPastMidnight;
+  }
+
+  public void setOpenPastMidnight(boolean openPastMidnight) {
+    this.openPastMidnight = openPastMidnight;
+  }
+
+  public CampusArea getArea() {
+    return mArea;
+  }
+
+  public void setArea(CampusArea area) {
+    this.mArea = area;
+  }
+
+
+  public enum Status {
+    OPEN,
+    CLOSING_SOON,
+    CLOSED;
+
+    public boolean isOpen() {
+      return this == OPEN || this == CLOSING_SOON;
+    }
+
+  }
+
+  /**
+   * Compared the time of two EateryModel
+   **/
+  public int compareTo(@NonNull EateryModel cm) {
+    if (cm.getCurrentStatus() == getCurrentStatus()) {
+      return this.getNickName().compareTo(cm.getNickName());
+    } else if (isOpen() && !cm.isOpen()) {
+      return -1;
+    } else {
+      return 1;
+    }
+  }
+
+  /*Comparator for sorting the list by EateryModel's nickname*/
+  public static Comparator<EateryModel> cafeNameComparator = new Comparator<EateryModel>() {
+    @Override
+    public int compare(EateryModel s1, EateryModel s2) {
+      String str1 = s1.getNickName();
+      String str2 = s2.getNickName();
+
+      // TODO Why?
```

Comment:
I'm pretty sure you added this TODO, but if this method isn't used, just delete it